inaugurator/partitiontable.py

```diff
@@ -267,90 +276,104 @@ def _isPartitionOfPhysicalDevice(self, device):
 
+    def _wipeOtherPartitionsWithSameVolumeGroup(self):
+        logging.info("Validating that volume group %(vg)s is bound only to one device..." % \
+              dict(vg=self.VOLUME_GROUP))
+        vgs = self._parseVGs()
+        if not vgs:
+            raise Exception("No volume group was found after configuration of LVM.")
+        targetPhysicalVolumeForVolumeGroup = self._getPartitionPath("lvm")
+        for physicalVolume, volumeGroup in vgs.iteritems():
+            if physicalVolume != targetPhysicalVolumeForVolumeGroup and volumeGroup == self.VOLUME_GROUP:
+                logging.info("Wiping '%(physicalVolume)s' since it contains another copy of the volume group..." \
+                      % dict(physicalVolume=physicalVolume))
+                self.clear(device=physicalVolume, count=1)
+                if self._isPartitionOfPhysicalDevice(physicalVolume):
+                    physicalDevice = self._getPhysicalDeviceOfPartition(physicalVolume)
+                    if physicalDevice == self._device:
+                        logging.info("Skipping wipe of the physical device that contained the volume group since" \
+                              " it's the target device.")
+                        continue
+                    logging.info("Wiping the physical device '%(physicalDevice)s' which contained the volume " \
+                          "group..." % dict(physicalDevice=physicalDevice))
+                    self.clear(device=physicalDevice, count=1)
+
     def verify(self):
-        if not self._findMismatch():
-            print "Partition table already set up"
+        mismatch = self._findMismatch()
+        if not mismatch:
+            logging.info("Partition table already set up")
             lvmPartitionPath = self._getPartitionPath("lvm")
             sh.run("lvm pvscan --cache %s" % lvmPartitionPath)
             for lv in ["root", "swap"]:
                 lv = "%s/%s" % (self.VOLUME_GROUP, lv)
-                print "Activating %s" % (lv,)
+                logging.info("Activating %s" % (lv,))
                 sh.run("lvm lvchange --activate y %s" % lv)
             sh.run("lvm vgscan --mknodes")
-            print "/dev/inaugurator:"
+            logging.info("/dev/inaugurator:")
             try:
-                print sh.run("busybox find /dev/inaugurator")
+                logging.info(sh.run("busybox find /dev/inaugurator"))
             except Exception as e:
-                print "Unable: %s" % e
-            self._wipeOtherPartitionsWithSameVolumeGroup()
-            self._wipeOtherPartitionsWithBootLabel()
+                logging.info("Unable: %s" % e)
+            self._wipeOldInstallationsIfAllowed()
             return
+        logging.warning("Found mismatch in partition layout - %s", mismatch)
+        if not self._wipeOldInstallations:
+            raise Exception("Found mismatch in partition layout. we cannot continue without wiping data")    
         self._create()
```

Comment:
you reach this point if you found a missmatch amd wipe is allowed. 
Doenst it make more sense first wipe and then create? Am I missing something?